src/parser.rs

```diff
@@ -884,6 +946,27 @@ impl Get for Attributes {
     }
 }
 
+fn read_text(parser: &mut Reader) -> Result<Option<String>, Error> {
+    let mut ret_text = None;
+
+    loop {
+        match try!(parser.next()) {
+            Characters(text) => {
+                ret_text = match ret_text {
+                    Some(t) => Some(format!("{}{}", t, text)),
+                    None => Some(text),
+                }
+            }
+            EndElement { .. } => break,
+            StartElement { name, .. } => return Err(mk_error!(&format!("Unexpected element: {}",
+                                                                       name),
```

Comment:
This is materially different from how other unexpected element errors are generated. `name` is not the same as `name.local_name`.